### ISSUE TITLE
[Advice] Add more information or comments to define_enum_for_matcher.rb?

### ISSUE THREAD

**Issue by festime** [thoughtbot/shoulda-matchers#927]
Hello, everyone. In my team's project, I called `define_enum_for` to validate the presence of some `enum`. I followed the style of code in comments of `define_enum_for_matcher.rb`, but I encountered some errors like this:

``` ruby
  1) Order validations with rails built-in methods should define :status as an enum with [:uncharged, :confirmed, :picking, :delivering, :finished, :refunding, :refunded, :cancel] and store the value in a column with an integer type
     Failure/Error:
       should define_enum_for(:status).with(
         [
           :uncharged, :confirmed, :picking, :delivering, :finished, :refunding,
           :refunded, :cancel
         ]


       Expected Order to define :status as an enum with [:uncharged, :confirmed, :picking, :delivering, :finished, :refunding, :refunded, :cancel] and store the value in a column with an integer type
```

My test code:

``` ruby
describe Order do
  describe "validations with rails built-in methods" do
    subject(:order) { build(:order) }

    it do
      should define_enum_for(:status).with(
        [
          :uncharged, :confirmed, :picking, :delivering, :finished, :refunding,
          :refunded, :cancel
        ]
      )
    end
  end

  # ...
end
```

My `Order` model

``` ruby
class Order < ActiveRecord::Base
  include OrderScopes

  enum status: {
    uncharged: 0, confirmed: 1, picking: 2, delivering: 3, finished: 4,
    refunding: 5, refunded: 6, cancel: 99
  }
  # ...
end
```

I read the source code of `DefineEnumForMatcher` and found that I can write:

``` ruby
describe Order do
  describe "validations with rails built-in methods" do
    subject(:order) { build(:order) }

    it do
      should define_enum_for(:status).with(
        uncharged: 0,
        confirmed: 1,
        picking: 2,
        delivering: 3,
        finished: 4,
        refunding: 5,
        refunded: 6,
        cancel: 99
      )
    end
  end
end
```

to make tests pass.

Maybe I can add some comments to `define_enum_for_matcher.rb` to make usages clearer?

**Comment by mcmire** [comment]
Sure, we can include an example of using `enum` with a hash of values in the documentation. We'd gladly accept a PR, too :)

**Comment by mcmire** [comment]
This is fixed: c47e6db7391963d27cb9b45c37d9aa94e53f0590
